frontend/src/app/item/item-cadastro/item-cadastro.component.ts

```diff
@@ -0,0 +1,148 @@
+import { ViewChild } from '@angular/core';
+import { ElementRef } from '@angular/core';
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { PageNotificationService } from '@nuvem/primeng-components';
+import { SelectItem } from 'primeng';
+import { finalize } from 'rxjs/operators';
+import { ItemService } from 'src/app/service/item.service';
+import { ItemModel } from '../model/item.model';
+
+@Component({
+  selector: 'app-item-cadastro',
+  templateUrl: './item-cadastro.component.html'
+})
+export class ItemCadastroComponent implements OnInit {
+
+  display: boolean = false;
+  submit: boolean = false;
+  isEditing: boolean = false;
+  form: FormGroup;
+  imagemUrl: string = "Selecione uma imagem";
+
+  @ViewChild('myImageInput')
+  myImageInputVariable: ElementRef;
+
+  @Output() itemListagemOutput: EventEmitter<void> = new EventEmitter();
+
+  categorias: SelectItem[] = [
+      {label: 'Colecionável', value: 1},
+      {label: 'Edição Especial', value: 2},
+      {label: 'Antiguidade', value: 3},
+      {label: 'Artesanal', value: 4},
+      {label: 'Memorabilia', value: 5},
+      {label: 'Item Autografado', value: 6},
+      {label: 'Outro', value: 7}
+  ];
+
+  constructor(
+    private itemService: ItemService, 
+    private fb: FormBuilder,
+    private notification: PageNotificationService
+    ) { }
+
+  iniciarForm() {
+    this.form = this.fb.group({
+      id: [null],
+      nome: [null, [Validators.required]],
+      descricao: [null, [Validators.required]],
+      foto: [null, [Validators.required]],
+      disponibilidade: [false, [Validators.required]],
+      situacao: ['OK', [Validators.required]],
+      idUsuario: [JSON.parse(localStorage.getItem("usuario")).id, [Validators.required]],
+      idCategoria: [null, [Validators.required]]
+    });
+  }
+
+  ngOnInit(): void {
+    this.itemListagemOutput.emit();
```

Comment:
realmente é necessário carregar os itens sempre que o modal for carregado?

Reply:
OK